pkg/rbac/parser.go

```diff
@@ -259,13 +259,107 @@ func GenerateRoles(ctx *genall.GenerationContext, roleName string) ([]interface{
 	return objs, nil
 }
 
+// GenerateRoleBindings generate a slice of objs representing either a ClusterRoleBinding or a RoleBinding object
+// according to the given roles
+func GenerateRoleBindings(roles []interface{}, roleName string) ([]interface{}, error) {
+	var objs []interface{}
+	for _, role := range roles {
+		if role, ok := role.(rbacv1.ClusterRole); ok {
+			objs = append(objs, rbacv1.ClusterRoleBinding{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ClusterRoleBinding",
+					APIVersion: rbacv1.SchemeGroupVersion.String(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: fmt.Sprintf("%sbinding", roleName),
+				},
+				Subjects: []rbacv1.Subject{
+					{
+						// keep consistent with kubebuilder's default
+						Kind:      rbacv1.ServiceAccountKind,
+						Name:      "controller-manager",
+						Namespace: "system",
+					},
+				},
+				RoleRef: rbacv1.RoleRef{
+					Kind:     role.Kind,
+					Name:     role.Name,
+					APIGroup: rbacv1.SchemeGroupVersion.Group,
+				},
+			})
+		}
+
+		if role, ok := role.(rbacv1.Role); ok {
+			objs = append(objs, rbacv1.RoleBinding{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "RoleBinding",
+					APIVersion: rbacv1.SchemeGroupVersion.String(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fmt.Sprintf("%sbinding", roleName),
+					Namespace: role.Namespace,
```

Comment:
Hi, I just got into the issue you refer to in this PR. Is the kustomize namespace replacement considered here? If I manually generate a role like this in my case, the namespace would be replaced with the one set in the main kustomization.yaml file.

Reply:
This 'controller-tools' is mainly used to parse the annotations of kubebuilder and generate the manifest, no kustomization.

Ref the kubebuilder, it do have kustomization, https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/common/kustomize/v2/scaffolds/internal/templates/config/rbac/kustomization.go.

If we want to overwrite the namespace, i prefer to use an additional config of https://kubectl.docs.kubernetes.io/references/kustomize/builtins/#_namespacetransformer.

Reply:
Are you stating that you would change how the namespace is set by the projects consuming controller-runtime (kube-builder, operator-sdk, ...)  as a follow-up that uses something like `unsetOnly: true` and `setRoleBindingSubjects: allServiceAccounts`?